src/components/Text/Text.tsx

```diff
@@ -1,49 +1,79 @@
-import React, { forwardRef } from "react";
+import clsx from "clsx";
+import React, { forwardRef, ForwardedRef } from "react";
+import styles from "./Text.module.css";
 
-import Typography, { TypographyProps } from "../common/typography";
+export type Size = "body" | "sm" | "md" | "lg" | "xs" | "caption" | "overline";
 
-type TextElement = "p" | "span";
+export type Variant =
+  | "error"
+  | "base"
+  | "brand"
+  | "inherit"
+  | "neutral"
+  | "success"
+  | "warning"
+  | "white"
+  /**
+   * @deprecated Info variant is deprecated.
+   */
+  | "info";
 
-type Props = {
+export type Props = {
   /**
    * Controls whether to render text inline (defaults to "p");
    */
-  as?: TextElement;
-  children: TypographyProps<TextElement>["children"];
-  className?: TypographyProps<TextElement>["className"];
-  color?: TypographyProps<TextElement>["color"];
-  size?: TypographyProps<TextElement>["size"];
-  spacing?: TypographyProps<TextElement>["spacing"];
+  as?: "p" | "span";
+  children: React.ReactNode;
+  className?: string;
+  variant?: Variant;
+  size?: Size;
   tabIndex?: number;
-  weight?: TypographyProps<TextElement>["weight"];
+  weight?: "bold" | "normal" | null;
 } & React.HTMLAttributes<HTMLElement>;
 
 /**
  * ```ts
  * import {Text} from "@chanzuckerberg/eds";
  * ```
  */
-const Text = forwardRef<HTMLElement, Props>(
+export const Text = forwardRef(
   (
     {
-      as,
+      as = "p",
       children,
+      className,
+      variant,
```

Comment:
ah, another breaking change we should note is that previously the typography default would set `color = "base"`, but now the default is essentially `inherit`. doubt that'll break anything, but good to note anyways

Reply:
I was thinking about this, should I be adding `variant="base"` to all the changes in LP?

Reply:
I don't think `variant="base"` was really that necessary 🤔  there's already a base color set on `body`, and I bet most text falls back to that. we can do a sanity check of the app to see if text colors change though